app/src/main/java/hacking/to/the/gate/GameManager.java

```diff
@@ -457,8 +501,74 @@ public void onTouchEvent(MotionEvent event) {
                 createSelfJet(event.getX(), event.getY());
             } else {
                 setSelfJetDest(event);
+                setOffBombs(event);
             }
         }
     }
 
+    /**
+     * Set the destination of the SelfJet.
+     * @param event MotionEvent of user touch.
+     */
+    private void setSelfJetDest(MotionEvent event){
+        switch (event.getAction())
+        {
+            case MotionEvent.ACTION_DOWN:
+            case MotionEvent.ACTION_MOVE:
+                if(mSelfJet!=null) {
+                    mSelfJet.setDestination(event.getX(), event.getY(), true);
+                }
+                break;
+            case MotionEvent.ACTION_UP:
+                if(mSelfJet!=null) {
+                    mSelfJet.setDestination(event.getX(), event.getY(), false);
+                }
+                break;
+        }
+    }
+    private void vibrateDevice(){
+        Vibrator v = (Vibrator)mContext.getSystemService(Context.VIBRATOR_SERVICE);
+        v.vibrate(60000);
+    }
+    private void stopVibration(){
+        Vibrator v = (Vibrator)mContext.getSystemService(Context.VIBRATOR_SERVICE);
+        v.cancel();
+    }
+    private void setOffBombs(MotionEvent event){
+        int pointerIndex = event.getActionIndex();
+        switch (event.getAction()& MotionEvent.ACTION_MASK){
+            case MotionEvent.ACTION_POINTER_DOWN:
+                Log.d("Touch","two fingers");
+                if(!mRemainingBombs.isEmpty()){
+                    int bombType = mRemainingBombs.get(0);
+                    mRemainingBombs.remove(0);
+                    float x = event.getX(pointerIndex);
+                    float y = event.getY(pointerIndex);
+                    switch (bombType){
+                        case 0:
+                            //x = getScreenRect().centerX();
+                            //y = getScreenRect().centerY();
+                            vibrateDevice();
+                            break;
+                        case 1:
+                            x = new Random().nextInt(100)+200;
+                            y = getScreenRect().centerY();
+                            break;
+                        case 2:
+                            y = getScreenRect().height();
+                            break;
+                        case 3:
+                            x = mSelfJet.getSelfPosition().getPositionX();
+                            y = mSelfJet.getSelfPosition().getPositionY();
+                            break;
+                    }
+                    synchronized (mBombs){
+                        mBombs.add(mBombFactory.getBomb(bombType,x,y,bombLifeCycle));
+                    }
```

Comment:
Does every bomb stop vibration when it's stopped? Better to put the lifecycle into BombFactory use anonymous class and call GameManager.getInstance().stopVibration()  when needed. @Jelly-Yu